ItemLevelDownloader/Store/artifactItemStore.ts

```diff
@@ -24,20 +24,26 @@ export class ArtifactItemStore {
         });
     }
 
+    public getTickets(): models.ArtifactDownloadTicket[] {
+        return this._downloadTickets;
+    }
+
     public getNextItemToProcess(): models.ArtifactItem {
         var nextItemToProcess = this._downloadTickets.find(x => x.state === models.TicketState.InQueue);
         if (nextItemToProcess) {
             nextItemToProcess.state = models.TicketState.Processing;
+            nextItemToProcess.startTime = new Date();
             return nextItemToProcess.artifactItem;
         }
 
         return undefined;
     }
 
-    public markAsProcessed(item: models.ArtifactItem) {
+    public updateState(item: models.ArtifactItem, state: models.TicketState) {
```

Comment:
looks like this is to update final state as it is updating the finish time unconditionally. final state can be skipped/processed/failed. in that case, rename to updateToFinalState and validate that the state passed is one of the final states?

Reply:
changed to updated finishtime only incase state is not queued or inprogress.